query/README.md

```diff
@@ -21,35 +21,183 @@ KeenQueryClient queryClient = new TestKeenQueryClientBuilder(queryProject)
-        .withRelativeTimeframe("this_month")
+        .withInterval("weekly")
+        .build();
+QueryResult result = queryClient.execute(query, new Timeframe("this_year"));
+if (result.isList()) {
+	ArrayList<QueryResult> listResults = result.getList();
+	for (QueryResult item : listResults) {
+		if (item.isInterval()) {
+			Interval interval = item.getInterval();
+			Timeframe itemTimeframe = interval.getTimeframe();
+			QueryResult intervalValue = interval.getValue();
+			if (intervalValue.isInteger()) {
+				// do something with integer result.
+			}
+		}
+	}
+}        
+```
+Filtering via both Group By and Interval will cause the query response to consist of an ArrayList of Interval objects. Each Interval object contains a Timeframe for the interval, as well as a QueryResult of the value. This QueryResult value will be another ArrayList of GroupBy objects, as follows:
+
+``` java
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
+        .withEventCollection("<event_collection>")
+        .withInterval("weekly")
         .withGroupBy("click-number")
         .build();
-Object result = queryClient.count(queryParams);
+QueryResult result = queryClient.execute(query, new Timeframe("this_year"));
+if (result.isList()) {
+    ArrayList<QueryResult> listResults = result.getList();
+    for (QueryResult item : listResults) {
+        if (item.isInterval()) {
+            Interval interval = item.getInterval();
+            Timeframe itemTimeframe = interval.getTimeframe();
+            QueryResult intervalValue = interval.getValue();
+            if (intervalValue.isList()) {
+                ArrayList<QueryResult> groupBys = intervalValue.getList();
+                for (QueryResult groupByItem : groupBys) {
+                    if (groupByItem.isGroupBy()) {
+                        GroupBy groupBy = groupByItem.getGroupBy();
+                        HashMap<String, Object> properties = groupBy.getProperties();
+                        QueryResult groupByResult = groupBy.getResult();
+                        if (groupByResult.isInteger()) {
+                            Integer val = groupByResult.getInteger();
+                                    // do something with integer result.
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
 ```
-There are also some utility methods to add filters and absolute timeframes to KeenQueryParams:
+
+### Utility Methods
+
+There are also some utility methods to add filters and absolute timeframes to a Query:
 ```java
-KeenQueryParams queryParams = new QueryParamBuilder()
+Query query = new QueryBuilder(QueryType.COUNT_RESOURCE)
 	            .withEventCollection(TEST_EVENT_COLLECTION)
 	            .build();
 
-queryParams.addFilter("click-count", "lt", 5);
-queryParams.addFilter("click-count", "gt", 1);
-queryParams.addAbsoluteTimeframe("2012-08-13T19:00:00.000Z", "2015-06-07T19:00:00.000Z");
+query.addFilter("click-count", "lt", 5);
```

Comment:
Question: Why not have these as part of the builder pattern. I would assume once an object is generated after calling `build()` it should be immutable.